istioctl/cmd/analyze.go

```diff
@@ -291,8 +309,12 @@ istioctl analyze -L
 
 			// Return code is based on the unfiltered validation message list/parse errors
 			// We're intentionally keeping failure threshold and output threshold decoupled for now
-			returnError := errorIfMessagesExceedThreshold(result.Messages)
-			if returnError == nil && parseErrors > 0 {
+			var returnError error
+			if messagesExceedThreshold(result.Messages) {
+				returnError = AnalyzerFoundIssuesError{namespace: selectedNamespace}
+			}
+
+			if parseErrors > 0 {
```

Comment:
This changes the semantics. Before if we have an AnalyzerFoundIssuesError we wouldn't use a parse error. Now we'll use a parse error regardless. This was based off of some discussion that we wanted to give some partial results back even if some of the files were malformed.

Reply:
I realize I left off the action item here, sorry :) If this was intentional then feel free to comment the reason and resolve. Otherwise I think not changing the semantics is better.

Reply:
Didn't realize this behavior was intentional. I backed it out.